ingester/src/data.rs

```diff
@@ -909,6 +991,25 @@ impl DataBuffer {
         Ok(())
     }
 
+    /// Return a QueryableBatch of the persisting batch after applying new tombstones
+    /// Only snapshots whose max_sequence_number > the given sequence_number are included
+    pub fn get_persisting_data(
+        &self,
+        sequence_number: Option<SequenceNumber>,
+    ) -> Result<Option<QueryableBatch>> {
+        if self.persisting.is_none() {
+            return Ok(None);
+        }
+
+        // persisting data
+        let mut queryable_batch = self.persisting.as_ref().unwrap().data.copy(sequence_number);
+
+        // Add new tombstones if any
+        queryable_batch.add_tombestones(&mut self.deletes_during_persisting.clone());
```

Comment:
Typo in this method name:

```suggestion
        queryable_batch.add_tombstones(&mut self.deletes_during_persisting.clone());
```